app/src/lib/trampoline/trampoline-askpass-handler.ts

```diff
@@ -62,7 +66,14 @@ async function handleSSHKeyPassphrase(
     return storedPassphrase
   }
 
-  const passphrase = await trampolineUIHelper.promptSSHKeyPassphrase(keyPath)
+  const {
+    passphrase,
+    storePassphrase,
+  } = await trampolineUIHelper.promptSSHKeyPassphrase(keyPath)
+
+  if (passphrase !== undefined && storePassphrase) {
+    keepSSHKeyPassphraseToStore(operationGUID, keyPath, passphrase)
```

Comment:
Another important bit. Now the UI prompt we show to the user just tells us if the user wanted to remember the passphrase, and if so we just "keep it around" in memory, associated to the `operationGUID` (which is just the trampoline token).